### ISSUE TITLE
Create GUI tests that run against Firefox and Chrome

### ISSUE THREAD

**Issue by lidel** [lidel/yt-looper#90]
I think we should go with [Nightwatch.js](http://nightwatchjs.org) as the test framework. It uses Selenium underneath and provides the [test runner](http://nightwatchjs.org/guide#test-runner)  as well as some useful [commands](http://nightwatchjs.org/api#commands) and [assertions](http://nightwatchjs.org/api#assertions).

The goal is to have [Travis CI](https://travis-ci.org/lidel/yt-looper) run tests against current Firefox and Chrome.
I do something similar for  [ipfs-firefox-addon](https://travis-ci.org/lidel/ipfs-firefox-addon) where I run against multiple Firefox versions.

As a PoC a simple Editor test should be created.

**Comment by lidel** [comment]
@sk4zuzu check this out:
https://travis-ci.org/lidel/yt-looper/builds/59316433

GUI tests can be run locally too. 
Initial Setup: 
```
npm install grunt-nightwatch --save-dev
bash test/nightwatch_dependency_check.sh
```

Then you can run against a browser of your choosing:

```
grunt nightwatch:firefox
grunt nightwatch:chrome
```

I plan to start writing GUI tests first and migrate headless tests from qunit to nightwatch in near future.

**Comment by sk4zuzu** [comment]
this looks fantastic! :+1: :rocket:

**Comment by lidel** [comment]
Short update: 

I managed to set up GUI tests under headless PhantomJS. 
This is much faster than running tests on real browser, 
so I've set it as a default target for both TravisCI and `grunt test`.

In future I'll see if I can set up real browsers at TravisCI somehow (in a way that gives stable builds), 
but for now it seems to be too slow for that. 
Let's keep in mind that this great service is provided for free. :+1: 

It is sill okay to run tests against real browsers locally.
I created handy aliases for two of them:

- `grunt firefox`
- `grunt chrome`

I also simplified everything related to tests and grunt, 
so if you are a new developer the only required step to run test suite is:

    npm install && npm test

It takes care of all dependencies, installs them to local directory 
named `node_modules` and runs `grunt test`.

Note that a simple HTTP server is spawned automatically for the lifetime of `npm test` task.
It is possible to run it in standalone mode via `grunt httpd`. It uses the domain whitelisted  by Google API, so that features such as URL shortener and video titles in Editor work the same way as in production.

See [CONTRIBUTING.md#tests](https://github.com/lidel/yt-looper/blob/master/CONTRIBUTING.md#tests) for more information.

I'll keep this ticket open for now to remind me to write GUI tests in spare time.
